boskos/boskos.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var configPath = flag.String("config", "projects.json", "Path to init projects file")
+var storage = flag.String("storage", "", "Path to presistent volume to load the state")
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+
+	r, err := NewRanch(*configPath, *storage)
+	if err != nil {
+		logrus.WithError(err).Fatal("Fail to create my ranch!")
+	}
+
+	http.Handle("/", handleDefault())
+	http.Handle("/request", handleRequestProject(r))
```

Comment:
So you want to make state generic and hope the user will do sane actions?

I'd rather make it a enum type of thing, so you can only call state.Next(), rather than set to a specific state.